### ISSUE TITLE
TypeError: Error #1034 cannot convert flash.events::ErrorEvent to Error.

### ISSUE THREAD

**Issue by mrfrasier** [DigitalStrawberry/ANE-Bugsnag#8]
Hey all,

Sorry to bring another issue up so soon 😄 

We use the cameraUI or cameraroll to get an image and display it on-stage. While tracking a bug reported by a user, I got this in the Animate console while following the reproduction steps:

> [BugsnagANE] TypeError: Error #1034: Type Coercion failed: cannot convert flash.events::ErrorEvent@10c72e0a1 to Error.

Here's my code:

> Bugsnag.getStackTrace = function(e:Error):String{ return e.getStackTrace(); }
			
> loaderInfo.uncaughtErrorEvents.addEventListener(UncaughtErrorEvent.UNCAUGHT_ERROR, Bugsnag.handleUncaughtError );

This is basically the exact code shown in the intro docs with me turnning the "error" variable to "e" instead.

So, if i'm reading this error right, it would seem the "e" that's getting passed to Bugsnag.getStackTrace is actually an ErrorEvent. So I changed my code to this, and it seems to work:

> Bugsnag.getStackTrace = function(e):String{
> 				if( e is Error ) return e.getStackTrace();
> 				else if( e is ErrorEvent ) return e.errorID + ": " + e.text;
> 				else return "What we got was unknown: ".concat( e );
> 			}

Now, I'm aware that "working" doesn't mean it's "right" haha. Did I do something wrong to hit that error in the first place? If not, any recommendations on improving my solution, or does it cover the known cases from collective experience?

**Comment by ecornbower** [comment]
@mrfrasier Thanks for reporting this, I'll have our developers look into it.

**Comment by marpies** [comment]
Hi @mrfrasier,

The `UncaughtErrorEvent` has `error` property which can be of any data type, though usually it's either `Error` or `ErrorEvent` - visit the [AS3 doc if you want to learn more about that](http://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/events/UncaughtErrorEvent.html#error).

Maybe I'm missing something but I don't understand why `Bugsnag.getStackTrace` is at fault. The value of the `error` property is passed to the `Bugsnag.getStackTrace` method, but [it is checked](https://github.com/DigitalStrawberry/ANE-Bugsnag/blob/master/actionscript/src/com/bugsnag/Bugsnag.as#L81-L85) whether it's actually of type `Error` before `Bugsnag.getStackTrace` is called. If the `ErrorEvent` is encountered, then the code must fail before `Bugsnag.getStackTrace` is called, though you say it's fixed when checking the data type within the stack trace method.

I forced an async event and the uncaught error reports `ErrorEvent` type as the data type of the `error` property but the stack trace method is never called, thus no error shows up. If you revert the stack trace method, can you reproduce the error message regularly? Also, are you sure you have no custom handlers for the `UncaughtErrorEvent`?

**Comment by mrfrasier** [comment]
As noted above, in your demo code, you have it typed to Error. In my code that works, I simply take a variable with no typing and check against it. Apparently an ErrorEvent was being passed instead of an Error.

**Comment by marpies** [comment]
We've updated the ANE to report any custom and async errors (previously ignored) caught by `handleUncaughtError` handler. The documentation regarding the stack trace method has been updated as well. We can only get the stack trace for synchronous errors so the code in the documentation handles only that. You shouldn't need to return any other 'stack trace' `String` for other error types, since all known information about the error will be reported in the `handleUncaughtError` handler.

Thanks for making us aware of this problem.
